### PR TITLE
Kb/create topics

### PR DESCRIPTION
**Pull request by kpsf**

### CODE REVIEW

File: src/main/java/com/salesforce/scmt/service/SalesforceService.java [hunk 1]
```diff
@@ -740,6 +812,32 @@ public static String createRemoteSite(Request req, Response res) throws Exceptio
...
+            if (e.getMessage().contains("INVALID_SESSION_ID")) {
+                res.status(401);
+                return "Unauthorized";
+            }
+        } catch(Exception e) {
+            // if (e.getMessage().contains("Remote Site Name already exists")) {
+            //     res.status(200);
+            //     return "Already Created";
+            // }
+            res.status(200);
```
Comment (tstachl) [inline src/main/java/com/salesforce/scmt/service/SalesforceService.java]: This should probably be a 400 or 500 status.

File: src/main/java/com/salesforce/scmt/service/SalesforceService.java [hunk 2]
```diff
@@ -740,6 +812,32 @@ public static String createRemoteSite(Request req, Response res) throws Exceptio
         return "Success";
     }
 
+    public static String createDataCategoryGroup(Request req, Response res) throws Exception {
+        String salesforceUrl = req.headers("Salesforce-Url");
+        String salesforceSessionId = req.headers("Salesforce-Session-Id");
+        System.out.println("made it to the base method");
```
Comment (tstachl) [inline src/main/java/com/salesforce/scmt/service/SalesforceService.java]: No need for this.

File: src/main/java/com/salesforce/scmt/service/SalesforceService.java [hunk 3]
```diff
@@ -157,6 +159,76 @@ public void createRemoteSite(RemoteSite rs)
...
+            if (r.isSuccess()) {
+                System.out.println("Created component: " + r.getFullName());
+            } else {
+                throw new Exception(r.getErrors()[0].getMessage());
+            }
+        }
+    }
+
+    public static DataCategory createDataCategory(DataCategoryJson dg) {
+        System.out.println("made it to create dc");
```
Comment (tstachl) [inline src/main/java/com/salesforce/scmt/service/SalesforceService.java]: not required

File: src/main/java/com/salesforce/scmt/service/SalesforceService.java [hunk 4]
```diff
@@ -157,6 +159,76 @@ public void createRemoteSite(RemoteSite rs)
         }
     }
 
+    public void createDataCategoryGroup(DataCategoryGroupJson dg)
+      throws ConnectionException, DeployException, AsyncApiException, Exception {
+        createMetadataConnection();
+        System.out.println("made it to create dcg");
```
Comment (tstachl) [inline src/main/java/com/salesforce/scmt/service/SalesforceService.java]: unnecessary

File: src/main/java/com/salesforce/scmt/service/SalesforceService.java [hunk 5]
```diff
@@ -157,6 +159,76 @@ public void createRemoteSite(RemoteSite rs)
...
+        dcg.setDescription(dg.description);
+        dcg.setActive(true);
+        dcg.setLabel(dg.label);
+        DataCategory dc = createDataCategory(dg.dataCategory);
+        dcg.setDataCategory(dc);
+        String[] listFullNames1 = new String[1];
+        List<String> listFullNames = new ArrayList<String>();
+        listFullNames.add(dg.fullName);
+        listFullNames.toArray( listFullNames1 );
+        System.out.println(listFullNames1);
```
Comment (tstachl) [inline src/main/java/com/salesforce/scmt/service/SalesforceService.java]: unnecessary

File: src/main/java/com/salesforce/scmt/service/SalesforceService.java [hunk 6]
```diff
@@ -157,6 +159,76 @@ public void createRemoteSite(RemoteSite rs)
...
+
+        com.sforce.soap.metadata.ReadResult existingResult = getMetadataConnection().readMetadata("DataCategoryGroup", listFullNames1);
+        System.out.println(existingResult);
```
Comment (tstachl) [inline src/main/java/com/salesforce/scmt/service/SalesforceService.java]: nope

File: src/main/java/com/salesforce/scmt/model/DataCategoryJson.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+ */
+
+
+package com.salesforce.scmt.model;
+
+public class DataCategoryJson {
+	public String name;
+	public String label;
+    public DataCategoryJson[] subCategories;
+    public DataCategoryJson parent;
```
Comment (tstachl) [inline src/main/java/com/salesforce/scmt/model/DataCategoryJson.java]: Can we format this correctly?

File: src/main/java/com/salesforce/scmt/model/DataCategoryGroupJson.java [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+
+package com.salesforce.scmt.model;
+import com.salesforce.scmt.model.DataCategoryJson;
+
+
+public class DataCategoryGroupJson {
+	public String fullName;
+	public String label;
+	public String description;
+    public DataCategoryJson dataCategory;
```
Comment (tstachl) [inline src/main/java/com/salesforce/scmt/model/DataCategoryGroupJson.java]: Formatting?

**Comment by kpsf** [comment]
@tstachl ok this is ready to review again

**Pull request merged**